cmake/spheral/SpheralHandleExt.cmake

```diff
@@ -0,0 +1,32 @@
+
+#----------------------------------------------------------------------------------------
+#                                   Spheral_Handle_Ext
+#----------------------------------------------------------------------------------------
+
+# -------------------------------------------
+# VARIABLES THAT NEED TO BE PREVIOUSLY DEFINED
+# -------------------------------------------
+# <lib_name>_DIR        : REQUIRED : The installation location of the TPL
+# <lib_name>_INCLUDES   : OPTIONAL : Specific includes for the TPL
+
+# ----------------------
+# INPUT-OUTPUT VARIABLES
+# ----------------------
+# <lib_name>     : REQUIRED : The name of the target TPL
+# TPL_CMAKE_DIR  : REQUIRED : Directory containing files for each TPL
+#                             listing their library names
+
+# -----------------------
+# OUTPUT VARIABLES TO USE - Made available implicitly after function call
+# -----------------------
+# <lib_name> : Exportable target for the TPL
+#----------------------------------------------------------------------------------------
+
+function(Spheral_Handle_Ext lib_name libname APPLE)
+
+  if(APPLE)
+    set(SHARED_EXT "dylib")
+    set(${lib_name}_libs ${libname}.${SHARED_EXT})
+  endif()
+
+endfunction()
```

Comment:
The documentation here doesn't match up with what this function is doing. This also seems to completely overwrites `${lib_name}_libs` to contain only a single library in the final list. I think a function that searches `${lib_name}_libs` for a set of libraries to be converted that finds and replaces them as necessary would be much more usefull.

Reply:
I've rewritten this function to replace .dylib or .so extensions with .a if ENABLE_STATIC_TPL=On.